
Epoch 0
Traceback (most recent call last):
  File "/Users/adamimos/Documents/GitHub/epsilon-transformers/./test.py", line 64, in sweep_train
    train_loss_per_position, train_loss_mean, test_loss_per_position, test_loss_mean = train_epoch(network, optimizer,
                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adamimos/Documents/GitHub/epsilon-transformers/./test.py", line 172, in train_epoch
    loss = criterion(outputs.view(-1, d_vocab), Y.view(-1), reduce='none') # (batch_size * input_size)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adamimos/anaconda3/envs/epsilon-machine/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adamimos/anaconda3/envs/epsilon-machine/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: CrossEntropyLoss.forward() got an unexpected keyword argument 'reduction'